feat: Store "published display name" and "published description" in search index [FC-0062]

ChrisChV opened the pull request:
<!--

Note: Please refer to the Support Development Guidelines on the wiki page to consider backporting to active releases:
https://openedx.atlassian.net/wiki/spaces/COMM/pages/4248436737/Support+Guidelines+for+active+releases

Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply.
You may link to information rather than copy it, but only if the link is publicly readable.
If the linked information must be private (because it contains secrets), clearly label the link as private.

-->

## Description

This store in the search index, the `display_name` and `description` of the published version of a library component.

- Which edX user roles will this change impact? "Course Author".

## Supporting information

Issue link: https://github.com/openedx/frontend-app-authoring/issues/1354
Internal ticket: [FAL-3881](https://tasks.opencraft.com/browse/FAL-3881)

## Testing instructions

- Create or use a library
- Add a component, and edit the content and the title.
- Go to [Meilisearch](http://meilisearch.local.edly.io:7700/) and search for your component. Verify the `display_name` and `description` and verify that it doesn't have a `published` dict.
- Publish the library.
- Return to Meilisearch and verify that the component has the `published` dict.
- Update the title and the content of the component.
- Return to Meilisearch and verify that the `display_name` and `description` have the new data, but the `published` has the old data.
- Publish the library
- Return to Meilisearch and verify the `published` has the new data.

## Deadline

ASAP

ChrisChV committed:
feat: Store "published display name" and "published description" in studio search index

openedx-webhooks commented:
Thanks for the pull request, @ChrisChV!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

ChrisChV committed:
test: Testing index for published blocks

ChrisChV committed:
style: Nits on the code

ChrisChV committed:
fix: Fix tests

bradenmacdonald commented on openedx/core/djangoapps/content/search/tests/test_documents.py:
```diff
@@ -297,6 +300,128 @@ def test_html_library_block(self):
+                "display_name": "Text 2",
+                "description": "This is a Test",
+            },
+        }
```
Nit: you don't have to check _all_ the fields of `doc` again here. You can just check `doc["display_name"]`, `doc["description"]`, and `doc["published"]`

bradenmacdonald reviewed (commented):
Thanks, this is great! I tested and it's working well.

Just one request: can you please add `published.display_name` and `published.description` to `update_searchable_attributes` in `edx-platform/openedx/core/djangoapps/content/search/api.py` ? Then we can do keyword search on them and get highlighting on those attributes, which will be useful in the "published only" version of the component picker.

bradenmacdonald committed:
feat: add published fields to the keyword searchable attributes list

bradenmacdonald reviewed (approved):
Good to go!

I rebased the PR for you and added those two fields to the keyword searchable list. I tested this out by rebuilding my own search index and by checking for those new fields in the search results JSON response.

ChrisChV commented:
> I rebased the PR for you and added those two fields to the keyword searchable list. I tested this out by rebuilding my own search index and by checking for those new fields in the search results JSON response.

Thanks!
